### PR TITLE
Importing: Give importing selective options #77

### PR DESCRIPTION
**Pull request by clyyuanzi-london**

### CODE REVIEW

**Comment by atbenmurray** [comment]
Are the include/exclude options on a per-table basis?

**Comment by clyyuanzi-london** [comment]
> Are the include/exclude options on a per-table basis?

Not right now, but can add that. Any suggestions on input argument examples? (maybe add --table-name ?)

**Comment by atbenmurray** [comment]
I think it needs to be, otherwise we get the name clashes across tables. I suggest that we have something like we do with the inputs 
```
patients:field1,field2,field3, assessments:field1,fieldfoo, bar
```
What do you think?

**Comment by clyyuanzi-london** [comment]
> I think it needs to be, otherwise we get the name clashes across tables. I suggest that we have something like we do with the inputs
> 
> ```
> patients:field1,field2,field3, assessments:field1,fieldfoo, bar
> ```
> 
> What do you think?

I think it's a good idea to follow the same format as the input.

**Comment by clyyuanzi-london** [comment]
> I think it needs to be, otherwise we get the name clashes across tables. I suggest that we have something like we do with the inputs
> 
> ```
> patients:field1,field2,field3, assessments:field1,fieldfoo, bar
> ```
> 
> What do you think?

I'm thinking if we can use pipe ( | ) instead of comma (,) to separate different tables.
```
patients:field1,field2,field3 | assessments: field1,foo, bar | ...
```

What do you think about this format?
Any suggestion on making it easier to use?

**Comment by atbenmurray** [comment]
You raise a valid point: what do you think of the following though?

```
patients:(field1,field2,field3), assessments:(field1,fieldfoo, bar)
vs.
patients:field1,field2,field3 | assessments:field1,fieldfoo, bar
```

**Comment by clyyuanzi-london** [comment]
Yeah, i agree parentheses makes more sense to users, compared to pipe. 

Just updated the code according to the suggested format:
```
patients:(field1,field2,field3), assessments:(field1,fieldfoo, bar)
```

**Comment by atbenmurray** [comment]
This is starting to look very good! I would suggest that it is possible for the user to set both --include and --exclude options, provided they do it for different tables. It makes sense that they may want to define the fields they want by blacklisting some tables but whitelisting others. What do you think?

**Comment by clyyuanzi-london** [comment]
I think it makes sense. Just one thing to clarify, if table A is listed in --include, then it shouldn't be listed in --exclude at the same time. What do you think?

**Comment by atbenmurray** [comment]
Agreed. It doesn't make sense to have include and exclude simultaneously for a given table. Two options:
1. show an error message and exit
2. apply include, then apply exclude (and probably warn)

What do you think?

**Comment by clyyuanzi-london** [comment]
I personally prefer the first option, as it can detect potential incorrect input.

**Comment by atbenmurray** [comment]
I'm happy with that if you are

**Comment by clyyuanzi-london** [comment]
Sounds good. I fixed following the first option then. Do you want to accept this PR?

File: exetera/bin/exetera [hunk 1]
```diff
@@ -28,62 +29,88 @@ except ModuleNotFoundError:
...
-    importer.import_with_schema(args.timestamp, args.output_hdf5, args.schema, tokens, args.overwrite)
+        if args.exclude:
+            isValid, exclude, exclude_table_names = validate_and_extract_arg_include_exclude(args.exclude) 
+            if not isValid:
+                raise ValueError(err_msg.format('-x/--exclude'))    
+
+        if args.include and args.exclude:
+            table_appear_twice = include_table_names & exclude_table_names
+            if len(table_appear_twice) > 0:
+                raise ValueError("'-n/--include, -x/--exclude': the following tables shouldn't listed both in -n/--include and -x/--exlude: {}"
```
[the remark this answers is not in the document]
Comment (clyyuanzi-london) [reply exetera/bin/exetera]: sounds good. just fixed

**Review by atbenmurray — APPROVED** [review]

**Pull request merged**